### PR TITLE
Refactors HydraHead so that version is defined in one place

### PR DESCRIPTION
**Pull request by reggieb**
In the current 9.2.2 release, version is defined in two places. 

- hydra-core/lib/hydra-head/version.rb
- HYDRA_VERSION

See discussion in https://github.com/projecthydra/hydra-head/issues/280

This code moves the definition of version to lib/hydra_head/version.rb

In making this change, I also noticed that the HydraHead module was defined in lib/hydra/head.rb. This does not comply to standard ruby practice and may cause loading issues in some environments. So I have also corrected this by moving the definition of module HydraHead from lib/hydra/head to lib/hydra_head. 

So as to provide backward compatibility, I've modified lib/hydra/head so that:

    require 'hydra/head'

Still loads HydraHead, but with a deprecation message.

### CODE REVIEW

**Comment by reggieb** [comment]
Look at the name spacing within hydra-core and hydra-access-controls, I'm not sure changing the file names to match the module was the right way to go. I wonder if the module name should match the file structure. That is, the project name space is Hydra, so I wonder if HydraHead should actually be Hydra::Head, and version be returned by Hydra::Head::VERSION and Hydra::Head.version

File: hydra-access-controls/hydra-access-controls.gemspec [hunk 1]
```diff
@@ -13,7 +14,7 @@ Gem::Specification.new do |gem|
   gem.test_files    = gem.files.grep(%r{^(test|spec|features)/})
   gem.name          = "hydra-access-controls"
   gem.require_paths = ["lib"]
-  gem.version       = version
+  gem.version       = HydraHead.version
```
Comment (jcoyne) [inline hydra-access-controls/hydra-access-controls.gemspec]: hydra-access-controls doesn't depend on hydra-head, hydra-head depends on hydra-access-controls.

File: hydra-core/hydra-core.gemspec [hunk 2]
```diff
@@ -13,13 +14,13 @@ Gem::Specification.new do |gem|
   gem.test_files    = gem.files.grep(%r{^(test|spec|features)/})
   gem.name          = "hydra-core"
   gem.require_paths = ["lib"]
-  gem.version       = version
+  gem.version       = HydraHead.version
```
Comment (jcoyne) [inline hydra-core/hydra-core.gemspec]: hydra-core doesn't depend on hydra-head. hydra-head depends on hydra-core

**Comment by jcoyne** [comment]
:-1: this change means you can't use hydra-core or hydra-access-controls without hydra-head.

**Comment by reggieb** [comment]
I don't think that's the case. The version is only used in building the gems. Once the gems are built, the code that is distributed in those gems does not need hydra-core. It's the same as the previous system - in that the gems didn't need to include the HYDRA_VERSION file.

**Comment by grosscol** [comment]
could the version.rb files all be pointed at the same HYDRA_VERSION file?

**Comment by jcoyne** [comment]
No. Because when they are packaged the sub-gems can't reference HYDRA_VERSION anymore.  I think the existing strategy is sound as it was adopted from Rails itself. https://github.com/rails/rails/blob/master/RAILS_VERSION

**Comment by grosscol** [comment]
To keep in line with the rails paradigm, we could change the version.rb files to their pattern.  That way the version would only be in defined in one location.
```ruby
require_relative 'gem_version'

module HydraHead
  # Returns the version of the currently loaded Hydra Head as a <tt>Gem::Version</tt>
  def self.version
    gem_version
  end
end
```

**Pull request closed without merging**